src/datanode/src/region_server.rs

```diff
@@ -140,6 +144,15 @@ impl RegionServer {
         region_id: RegionId,
         request: RegionRequest,
     ) -> Result<RegionResponse> {
+        if let Some(semaphore) = &self.inner.concurrency_limit {
+            let permit = semaphore
+                .acquire()
+                .await
+                .context(OverConcurrencyLimitSnafu)?;
```

Comment:
we can add a configurable timeout here to make it more usable, code like

```
select!{
  semaphore.acquire(), timeout(...)
}
```

Reply:
is the timeout act like a lock timeout for acquiring permit from semaphore?

Reply:
Yes. Comparing to plain `acquire`, this allows request to fail without waiting forever.